contracts/src/main/kotlin/com/contractsAndStates/states/LongestRoadState.kt

```diff
@@ -0,0 +1,15 @@
+package com.contractsAndStates.states
+
+import com.contractsAndStates.contracts.LongestRoadContract
+import net.corda.core.contracts.BelongsToContract
+import net.corda.core.contracts.LinearState
+import net.corda.core.contracts.UniqueIdentifier
+import net.corda.core.identity.AbstractParty
+import net.corda.core.identity.Party
+import net.corda.core.serialization.CordaSerializable
+
+@CordaSerializable
+@BelongsToContract(LongestRoadContract::class)
+data class LongestRoadState(val holder: Party?,
```

Comment:
Why not `val holder: Party`. After all, if there is no holder, then there should be no state.

Reply:
I did it this way to handle the following use case:
1. Player 1 receives the card after building a 5 segment road
1. Player 2 builds a settlement in the middle of the longest road of the other player
1. No other player meets the requirement of Longest Road
1. The card goes back to the "bank"

Reply:
Makes sense.

Reply:
Sounds good to me. Good catch.